tests/unit/test_client.py

```diff
@@ -1784,3 +1792,50 @@ def test_get_signed_policy_v4_with_access_token(self):
             policy["fields"]["policy"],
             b"eyJjb25kaXRpb25zIjpbeyJidWNrZXQiOiJidWNrZXQtbmFtZSJ9LHsiYWNsIjoicHJpdmF0ZSJ9LFsic3RhcnRzLXdpdGgiLCIkQ29udGVudC1UeXBlIiwidGV4dC9wbGFpbiJdLHsia2V5Ijoib2JqZWN0LW5hbWUifSx7IngtZ29vZy1kYXRlIjoiMjAyMDAzMTJUMTE0NzE2WiJ9LHsieC1nb29nLWNyZWRlbnRpYWwiOiJ0ZXN0QG1haWwuY29tLzIwMjAwMzEyL2F1dG8vc3RvcmFnZS9nb29nNF9yZXF1ZXN0In0seyJ4LWdvb2ctYWxnb3JpdGhtIjoiR09PRzQtUlNBLVNIQTI1NiJ9XSwiZXhwaXJhdGlvbiI6IjIwMjAtMDMtMjZUMDA6MDA6MTBaIn0=",
         )
+
+
+@pytest.mark.parametrize("test_data", _POST_POLICY_TESTS)
+def test_conformance_post_policy(test_data):
+    import datetime
+    from google.cloud.storage.client import Client
+
+    client = Client(credentials=_DUMMY_CREDENTIALS)
+
+    in_data = test_data["policyInput"]
+    out_data = test_data["policyOutput"]
+    out_fields = test_data["policyOutput"]["fields"]
+
+    bucket_bound_hostname = None
+    if in_data.get("urlStyle") == "BUCKET_BOUND_HOSTNAME":
+        bucket_bound_hostname = in_data["bucketBoundHostname"]
+
+    timestamp = datetime.datetime.strptime(in_data["timestamp"], "%Y-%m-%dT%H:%M:%SZ")
+
+    with mock.patch("google.cloud.storage._signing.NOW", return_value=timestamp):
+        with mock.patch(
+            "google.cloud.storage.client.get_expiration_seconds_v4",
+            return_value=in_data["expiration"],
+        ):
+            with mock.patch("google.cloud.storage.client._NOW", return_value=timestamp):
+                policy = client.generate_signed_post_policy_v4(
+                    credentials=_DUMMY_CREDENTIALS,
+                    bucket_name=in_data["bucket"],
+                    blob_name=in_data["object"],
+                    conditions=in_data.get("conditions"),
```

Comment:
how does this transform dictionary into list?

https://github.com/googleapis/conformance-tests/blob/master/storage/v1/v4_signatures.json#L401

Reply:
Yeah, that's something I've asked in the chatroom. Pushed